plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java

```diff
@@ -1835,6 +1835,116 @@ public void testIncorrectIcebergFileSizes()
         dropTable("test_iceberg_file_size");
     }
 
+    @Test
+    public void testJoinWithPartitionTable()
+    {
+        // non partitioned tables
+        assertUpdate("CREATE TABLE test_join_partitioning_normal1(a BIGINT, b VARCHAR)");
+        assertUpdate("CREATE TABLE test_join_partitioning_normal2(a BIGINT, b VARCHAR, c row(d BIGINT, e VARCHAR), g map(BIGINT, VARCHAR))");
+
+        String normalValues1 = "VALUES (0, 'a'), (1,'b'), (2, 'c'), (3, 'd'),(4, NULL)";
+        String normalValues2 = "VALUES " +
+                "(0, 'a', (0, 'a'), map(ARRAY[1], ARRAY['a']))," +
+                "(1, 'b', (1,'b'), map(ARRAY[1], ARRAY['b'])), " +
+                "(2, 'c', (2, 'c'), map(ARRAY[2], ARRAY['c'])), " +
+                "(3, 'd', (3, 'd'), map(ARRAY[3, 33], ARRAY['d', 'x']))," +
+                "(4, NULL, NULL, NULL), " +
+                "(5, 'f', (5, NULL), map()) ";
+        assertUpdate("INSERT INTO test_join_partitioning_normal1 " + normalValues1, 5);
+        assertUpdate("INSERT INTO test_join_partitioning_normal2 " + normalValues2, 6);
+
+        String expectedValues1 = "VALUES (1,'b'), (2, 'c'), (3, 'd'),(4, NULL)";
+        String expectedvalues2 = "VALUES " +
+                "(1, 'b'), " +
+                "(2, 'c'), " +
+                "(3, 'd')," +
+                "(4, NULL), " +
+                "(5, 'f') ";
+
+        // partitioned tables
```

Comment:
please initialize the tables with data in this section. And do only assertions in following sections.